include/highfive/bits/H5Slice_traits_misc.hpp

```diff
@@ -88,36 +90,32 @@ SliceTraits<Derivate>::select(const std::vector<size_t>& offset,
     if (H5Sselect_hyperslab(space.getId(), H5S_SELECT_SET, offset_local.data(),
                             stride.empty() ? NULL : stride_local.data(),
                             count_local.data(), NULL) < 0) {
-        HDF5ErrMapper::ToException<DataSpaceException>(
-            "Unable to select hyperslap");
+        HDF5ErrMapper::ToException<DataSpaceException>("Unable to select hyperslap");
     }
 
     return Selection(DataSpace(count), space,
                      details::get_dataset(static_cast<const Derivate*>(this)));
 }
 
 template <typename Derivate>
-inline Selection
-SliceTraits<Derivate>::select(const std::vector<size_t>& columns) const {
-
-    const DataSpace& space = static_cast<const Derivate*>(this)->getSpace();
-    const DataSet& dataset =
-        details::get_dataset(static_cast<const Derivate*>(this));
+inline Selection SliceTraits<Derivate>::select(const std::vector<size_t>& columns) const {
+    const auto* slice = static_cast<const Derivate*>(this);
+    const DataSpace& space = slice->getSpace();
+    const DataSet& dataset = details::get_dataset(slice);
     std::vector<size_t> dims = space.getDimensions();
     std::vector<hsize_t> counts(dims.size());
     std::copy(dims.begin(), dims.end(), counts.begin());
     counts[dims.size() - 1] = 1;
     std::vector<hsize_t> offsets(dims.size(), 0);
 
     H5Sselect_none(space.getId());
-    for (std::vector<size_t>::const_iterator i = columns.begin();
-         i != columns.end(); ++i) {
+    for (std::vector<size_t>::const_iterator i = columns.begin(); i != columns.end();
```

Comment:
for range based loop here

Reply:
we still to make a global c++11 modernization... I only "fixed" the line splitting